Source/ChocolateyGui/Providers/ChocolateyConfigurationProvider.cs

```diff
@@ -7,65 +7,80 @@
+        private readonly Lazy<bool> _isChocolateyExecutableBeingUsed;
 
         public ChocolateyConfigurationProvider()
         {
-            this.GetChocolateyInstallLocation();
-            this.DetermineIfChocolateyExecutableIsBeingUsed();
+            this._chocolateyInstallLocation = new Lazy<string>(this.GetChocolateyInstallLocation);
+            this._isChocolateyExecutableBeingUsed = new Lazy<bool>(this.DetermineIfChocolateyExecutableIsBeingUsed);
         }
 
         public string ChocolateyInstall
         {
-            get
-            {
-                return this._chocolateyInstallLocation;
-            }
+            get { return this._chocolateyInstallLocation.Value; }
         }
 
         public bool IsChocolateyExecutableBeingUsed
         {
-            get
-            {
-                return this._isChocolateyExecutableBeingUsed;
-            }
+            get { return this._isChocolateyExecutableBeingUsed.Value; }
+        }
+
+        public IEnumerable<Tuple<string, string>> Sources
+        {
+            get { return this.GetChocolateySources(); }
         }
 
-        private void GetChocolateyInstallLocation()
+        private string GetChocolateyInstallLocation()
         {
-            this._chocolateyInstallLocation = Environment.GetEnvironmentVariable("ChocolateyInstall");
-            if (string.IsNullOrWhiteSpace(this._chocolateyInstallLocation))
+            var retVal = Environment.GetEnvironmentVariable("ChocolateyInstall");
+            if (string.IsNullOrWhiteSpace(retVal))
             {
                 var pathVar = Environment.GetEnvironmentVariable("PATH");
                 if (!string.IsNullOrWhiteSpace(pathVar))
                 {
-                    this._chocolateyInstallLocation =
+                    retVal =
                         pathVar.Split(';')
                             .SingleOrDefault(
                                 path => path.IndexOf("Chocolatey", StringComparison.OrdinalIgnoreCase) > -1);
                 }
             }
 
-            if (string.IsNullOrWhiteSpace(this._chocolateyInstallLocation))
+            if (string.IsNullOrWhiteSpace(retVal))
             {
-                this._chocolateyInstallLocation = string.Empty;
+                retVal = string.Empty;
             }
+
+            return retVal;
         }
 
-        private void DetermineIfChocolateyExecutableIsBeingUsed()
+        private bool DetermineIfChocolateyExecutableIsBeingUsed()
         {
-            var exePath = Path.Combine(this._chocolateyInstallLocation, "choco.exe");
+            var exePath = Path.Combine(this._chocolateyInstallLocation.Value, "choco.exe");
+
+            return File.Exists(exePath);
+        }
 
-            if (File.Exists(exePath))
+        private IEnumerable<Tuple<string, string>> GetChocolateySources()
+        {
+            var chocoConfigFile = Path.Combine(this._chocolateyInstallLocation.Value, "config", "chocolatey.config");

```

Comment:
Completely agree. I didn't see such a capability, or know of one. Do point me in that direction if there is one - would love to play with it (and use it here of course) :)